### ISSUE TITLE
Crash from FastColoredTextBox

### ISSUE THREAD

**Issue by amweiss** [amweiss/vigilant-cupcake#32]
Happened while it was in the background, possibly from a download/resave?

```
************** Exception Text **************
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException()
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at FastColoredTextBoxNS.TextSource.get_Item(Int32 i)
   at FastColoredTextBoxNS.FastColoredTextBox.get_Item(Int32 iLine)
   at FastColoredTextBoxNS.Range.GetText(String& text, List`1& charIndexToPlace)
   at FastColoredTextBoxNS.Range.<GetRanges>d__0.MoveNext()
   at FastColoredTextBoxNS.Range.SetStyle(StyleIndex styleLayer, String regexPattern, RegexOptions options)
   at FastColoredTextBoxNS.Range.SetStyle(Style style, String regexPattern)
   at VigilantCupcake.ViewUtilities.FastColoredTextBoxUtility.setStyles(Range range)
   at VigilantCupcake.MainForm.<updateHostsFileView>b__48()
   at System.Windows.Forms.Control.InvokeMarshaledCallbackDo(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Forms.Control.InvokeMarshaledCallback(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbacks()
```

**Comment by amweiss** [comment]
Hopefully fixed in c4248ca9a2b09a2f36bc968332d48c4c844ebd47

**Comment by amweiss** [comment]
Still happening it seems

**Comment by amweiss** [comment]
It appears that it happens when a remote source is updated while enabled and/or selected.

**Comment by amweiss** [comment]
Really should be fixed now in e3209a3800dca77dd800c1ff2fa6c5448d4513d4 (I hope).
